WordPress/src/main/java/org/wordpress/android/WordPress.java

```diff
@@ -280,6 +289,66 @@ public void onCreate() {
         AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
     }
 
+    private void migrateAccessToken() {
+        // Migrate access token AccountStore
+        if (!AppPrefs.wasAccessTokenMigrated() && !mAccountStore.hasAccessToken()) {
+            AppLog.i(T.DB, "No access token found in FluxC - attempting to migrate existing one");
+            // It will take some time to update the access token in the AccountStore if it was migrated
+            // so it will be set to the migrated token
+            String migratedToken = WPLegacyMigrationUtils.migrateAccessTokenToAccountStore(this, mDispatcher);
+            if (!TextUtils.isEmpty(migratedToken)) {
+                AppLog.i(T.DB, "Access token successfully migrated to FluxC - fetching accounts and sites");
+                AppPrefs.setAccessTokenMigrated(true);
+
+                mDispatcher.dispatch(AccountActionBuilder.newFetchAccountAction());
+                mDispatcher.dispatch(AccountActionBuilder.newFetchSettingsAction());
+                mDispatcher.dispatch(SiteActionBuilder.newFetchSitesAction());
+                return;
+            }
+            // Even if there was no token to migrate, turn this flag on so we don't attempt to migrate again
+            AppPrefs.setAccessTokenMigrated(true);
+        }
+
+        migrateSelfHostedSites();
+    }
+
+    private void migrateSelfHostedSites() {
+        if (!AppPrefs.isSelfHostedSitesMigratedToFluxC()) {
+            List<SiteModel> siteList = WPLegacyMigrationUtils.migrateSelfHostedSitesFromDeprecatedDB(this, mDispatcher);
+            if (siteList != null && !siteList.isEmpty()) {
+                AppLog.i(T.DB, "Finished migrating " + siteList.size() + " self-hosted sites - fetching site info");
+                mRemainingSelfHostedSitesToFetch = siteList.size();
+                for (SiteModel siteModel : siteList) {
+                    mDispatcher.dispatch(SiteActionBuilder.newFetchSiteAction(siteModel));
+                }
+            } else {
+                AppLog.i(T.DB, "No self-hosted sites to migrate");
+                endMigration();
+            }
+            AppPrefs.setSelfHostedSitesMigratedToFluxC(true);
+        } else {
+            AppLog.i(T.DB, "Self-hosted sites have already been migrated");
+            endMigration();
+        }
+    }
+
+    private void endMigration() {
+        AppLog.i(T.DB, "Ending migration to FluxC");
+        sIsMigrationInProgress = false;
+        if (sMigrationListener != null) {
+            sMigrationListener.onCompletion();
+            sMigrationListener = null;
+        }
+    }
+
+    public static void registerMigrationListener(MigrationListener listener) {
+        if (!sIsMigrationInProgress) {
+            listener.onCompletion();
```

Comment:
That was necessary to immediately dismiss the progress dialog and load the UI, since I wasn't exposing another method to check if migration was in progress. (I think from your next comment this became clear.)